src/pages/index.js

```diff
@@ -27,71 +32,185 @@ const userinfoComponent = new Userinfo(
+);
+deleteCardPopup.setEventListeners();
+function handleDeleteClick(card, cardID) {
+  deleteCardPopup.setSubmitAction(() => {
+    deleteCardPopup.showLoading();
+    api
+      .deleteCard(cardID)
+      .then(() => {
+        card.deleteCard();
+        deleteCardPopup.close();
+      })
+      .catch((err) => console.error(err))
+      .finally(() => {
+        deleteCardPopup.hideLoading();
+      });
+  });
+
+  deleteCardPopup.open();
 }
 
-const section = new Section(
-  {
-    items: initialCards,
-    renderer: renderCard,
+// Editing Profile //
+const editAvatarPopup = new PopupWithForm(
+  "#edit-avatar-modal",
+  (inputValues) => {
+    editAvatarPopup.renderLoading();
+
+    api
+      .editProfileImage(inputValues)
+      .then(() => {
+        profileAvatar.src = inputValues.link;
+        editAvatarPopup.close();
+      })
+
+      .catch((err) => console.error(err))
+      .finally(() => {
+        editAvatarPopup.renderLoading();
+      });
   },
-  containerSelector
+  "Saving..."
 );
 
-section.renderItems();
+profileAvatarButton.addEventListener("click", () => {
+  editAvatarPopup.open();
+});
+
+editAvatarPopup.setEventListeners();
 
-// Modal Image
-const imagePreviewModal = new PopupWithImage(imageModalSelector);
-imagePreviewModal.setEventListeners();
+// Likes///
+function handleLikeClick(card) {
+  if (card.isLiked()) {
+    api
+      .likeCountRemove(card)
+      .then((updatedCard) => {
+        card.setLikes(updatedCard.likes);
+      })
+      .catch((error) => {
+        console.error("Error removing like:", error);
+      });
+  } else {
+    api
+      .likeCountAdd(card)
+      .then((updatedCard) => {
+        card.setLikes(updatedCard.likes);
+      })
+      .catch((error) => {
+        console.error("Error adding like:", error);
+      });
+  }
+}
+
+const section = new Section({}, containerSelector);
 
 // Functions
+
+function renderCard(cardData) {
+  const card = new Card(
+    cardData,
+    "#card-template",
+    handleCardImageClick,
+    (cardID) => handleDeleteClick(card, cardID),
+    handleLikeClick
+  );
+
+  return card.getView();
+}
+
 function addCard(data) {
   const cardData = {
     name: data.title,
     link: data.imageURL,
   };
 
-  renderCard(cardData);
+  const cardElement = renderCard(cardData);
+  section.addItem(cardElement);
+
   addCardFormPopup.close();
+
+  api
+    .addNewCard(cardData)
+    .then((newCardData) => {
+      renderCard(newCardData);
+
+      userinfoComponent.setUserImage(newCardData.avatar);
+      imagePreviewModal.close();
+    })
+    .catch((error) => {
+      console.error("Error adding new card:", error);
+    })
+    .finally(() => {});
 }
 
+const api = new Api(apiURL);
+
+const imagePreviewModal = new PopupWithImage(imageModalSelector);
+imagePreviewModal.setEventListeners();
+
+function handleCardImageClick(cardData) {
+  imagePreviewModal.open(cardData);
+}
+
+api.getInitialCards().then((cardData) => {
+  cardData.forEach((cardItem) => {
```

Comment:
Please remove lines 188-200. `section.renderItems();` should accept `items` as an argument and render them

Reply:
Please fix this issue

Reply:
Please fix this issue

Reply:
Please remove this loop, Section class should render the items, like in project 8

Reply:
Please fix this issue